### PR TITLE
Make sure build server is connected or not available before parsing

### PR DESCRIPTION
**Pull request by tgodzik**
Fixes https://github.com/scalameta/metals/issues/2094

### CODE REVIEW

**Review by ckipp01 — APPROVED** [review]
This is a much smaller change than I thought it was going to take for this, but yea, this make total sense. LGTM!

**Review by ckipp01 — CHANGES_REQUESTED** [review]
Actually the referenceLsp suite seems to not be happy.

File: tests/unit/src/test/scala/tests/ReferenceLspSuite.scala [hunk 1]
```diff
@@ -38,11 +38,11 @@ class ReferenceLspSuite extends BaseRangesSuite("reference") {
           |  val z: Int = y.a
           |  val param: String = a.A.param(arg = 2)
           |}
-          |""".stripMargin,
-        preInitialized = { () => server.didOpen("a/src/main/scala/a/A.scala") }
```
Comment (tgodzik) [inline tests/unit/src/test/scala/tests/ReferenceLspSuite.scala:42]: So this would hang, because it's run before sending `initialized` message.

Comment (ckipp01) [reply tests/unit/src/test/scala/tests/ReferenceLspSuite.scala:42]: Ah gotcha, that makes sense.

**Review by ckipp01 — APPROVED** [review]
Looks good! LGTM!

**Comment by tgodzik** [comment]
JFR is failing again, which makes no sense :/ I will take a look at it later on.

**Pull request merged**